Trippy/API/LogoutHandler.m

```diff
@@ -0,0 +1,23 @@
+//
+//  LogoutHandler.m
+//  Trippy
+//
+//  Created by Catherine Lu on 7/7/22.
+//
+
+#import "LogoutHandler.h"
+#import "Parse/Parse.h"
+
+@implementation LogoutHandler
+
+- (void) logoutCurrentUser {
+    [PFUser logOutInBackgroundWithBlock:^(NSError * _Nullable error) {
+        if(error) {
+            [self.delegate logoutFail:error];
+        } else {
+            [self.delegate logoutSuccess];
+        }
```

Comment:
I thought the same, but since some delegate methods wouldn't need to be implemented in each file (ie. login only used in the login screen, whereas logout wouldn't be), I wasn't sure if it made sense to put them under the same protocol. Is it possible to not implement all protocol methods / create multiple protocols within the same class?

Reply:
I see what you mean. You could technically use `@optional` for them, but then at the caller you'd have to wrap everything with `respondsToSelector:` checks and that's really messy. If you want to keep with delegates and not blocks, maybe put both delegates in the `UserAuthHandler` (I still think that's the best place), and rename the other one `LogInAndRegisterHandlerDelegate` or some such.